R/get_source_expressions.R

```diff
@@ -93,14 +95,23 @@ get_source_expressions <- function(filename, lines = NULL) {
       top_level_map
     )
 
+    expression_xmls <- lapply(
+      xml2::xml_find_all(xml_parsed_content, "/exprlist/*"),
+      function(top_level_expr) xml2::xml_add_parent(xml2::xml_new_root(top_level_expr), "exprlist")
```

Comment:
This is the magic 😄  (faster than parsing a new XML is copying the node from the existing parsed XML and adding the `exprlist` parent node.

Reply:
does this actually make a copy / are our XML trees always read-only?

i have this in mind:

https://github.com/r-lib/xml2/issues/341

Reply:
(awesome idea btw & makes perfect sense to me, just trying to stress test it)

Reply:
`xml_new_root()` has a `.copy` argument, which defaults to `TRUE` in our usecase. This ensures a new document is created with a copy of that particular selected node as the root element.